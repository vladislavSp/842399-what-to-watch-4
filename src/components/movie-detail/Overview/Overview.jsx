import React from 'react';
import PropTypes from 'prop-types';

const Overview = ({filmContent}) => (<React.Fragment>
  <div className="movie-rating">
    <div className="movie-rating__score">{filmContent.rating}</div>
    <p className="movie-rating__meta">
      <span className="movie-rating__level">-</span>
      <span className="movie-rating__count">{10} ratings</span>
    </p>
  </div>

  <div className="movie-card__text">
    <p>{filmContent.description}</p>

    <p className="movie-card__director"><strong>Director: {filmContent.producer}</strong></p>

    <p className="movie-card__starring"><strong>Starring: {filmContent.actors.map((el) => `${el}, `) } and other</strong></p>
  </div>
</React.Fragment>);

Overview.propTypes = {
  filmContent: PropTypes.object.isRequired,
};

export default Overview;
