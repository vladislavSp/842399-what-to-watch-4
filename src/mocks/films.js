const films = [
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`,
    genre: `Comedies`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Macbeth`,
    genre: `Comedies`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Aviator`,
    genre: `Dramas`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `We need to talk about Kevin`,
    genre: `Thrillers`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Shutter Island`,
    genre: `Horror`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `No Country for Old Men`,
    genre: `Romance`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Snatch`,
    genre: `Sci-Fi`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

const promoFilm = {
  promoTitle: `The Grand Budapest Hotel`,
  promoGenre: `Comedy/Drama`,
  promoYear: 2014,
};

export {films, promoFilm};
