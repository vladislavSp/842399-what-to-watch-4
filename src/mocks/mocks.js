const films = [
  {
    id: 0,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 1,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 3,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 5,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 6,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 7,
    title: `Gangs of new york`,
    producer: `Martin Scorsese`,
    genre: `Crime`,
    year: 2002,
    rating: `8.8`,
    actors: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`
    ],
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

const promoFilm = {
  id: `1`,
  title: `The Grand Budapest Hotel`,
  genre: `Comedy/Drama`,
  year: 2014,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`
};

const filmContent = {
  title: `Gangs of new york`,
  producer: `Martin Scorsese`,
  genre: `Crime`,
  year: 2002,
  rating: `8.8`,
  actors: [
    `Leonardo DiCaprio`,
    `Cameron Diaz`,
    `Daniel Day-Lewis`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`
  ],
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  background: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

const reviews = [
  {
    id: 0,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    rating: `8,9`
  },
  {
    id: 1,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    rating: `8,9`
  }
];

export const userObj = {
  id: `1`,
  avatarUrl: `avatar`,
  email: `email`,
  name: `name`
};

export {films, promoFilm, filmContent, reviews};
