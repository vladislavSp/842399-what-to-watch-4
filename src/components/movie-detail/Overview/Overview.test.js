import React from 'react';
import renderer from 'react-test-renderer';
import {filmContent} from '../../../mocks/films';
import Overview from './Overview.jsx';

it(`VideoPlayer render correctly`, () => {
  const tree = renderer.create((
    <Overview
      filmContent={filmContent}
    />
  ), {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
