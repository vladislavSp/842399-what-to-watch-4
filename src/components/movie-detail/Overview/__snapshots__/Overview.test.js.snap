// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer render correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        -
      </span>
      <span
        className="movie-rating__count"
      >
        10
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Martin Scorsese
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Leonardo DiCaprio, 
        Cameron Diaz, 
        Daniel Day-Lewis, 
        Jude Law, 
        Willem Dafoe, 
        Saoirse Ronan, 
         and other
      </strong>
    </p>
  </div>,
]
`;
